custom_components/places/sensor.py

```diff
@@ -909,15 +914,23 @@ def get_zone_details(self):
                 + str(self.get_attr(ATTR_DEVICETRACKER_ZONE_NAME))
             )
         else:
+            name = self.get_attr(ATTR_DEVICETRACKER_ZONE)
+            if self._hass.states.get(
+                "zone." + name
+            ) is not None and self._hass.states.get(
+                    "zone." + name
+                ).attributes.get(CONF_FRIENDLY_NAME) is not None:
+                    name = self._hass.states.get(
+                        "zone." + name
+                    ).attributes.get(CONF_FRIENDLY_NAME)
             _LOGGER.debug(
                 "("
                 + self.get_attr(CONF_NAME)
                 + ") DeviceTracker Zone: "
-                + str(self.get_attr(ATTR_DEVICETRACKER_ZONE))
+                + str(name)
             )
             self.set_attr(
-                ATTR_DEVICETRACKER_ZONE_NAME, self.get_attr(ATTR_DEVICETRACKER_ZONE)
-            )
+                ATTR_DEVICETRACKER_ZONE_NAME, name)
 
```

Comment:
Here, the tracker isn't in a zone so it just reports the state of the tracker. Since it's not in a zone, I don't think the if would ever be true and thus it's not needed